crates/sncast/src/lib.rs

```diff
@@ -327,6 +328,54 @@ fn get_account_from_accounts_file<'a>(
     Ok(account)
 }
 
+pub async fn check_if_legacy_contract(
+    class_hash: Option<FieldElement>,
+    address: FieldElement,
+    provider: &JsonRpcClient<HttpTransport>,
+) -> Result<bool> {
+    let class_hash = extract_class_hash_or_fetch(class_hash, address, provider).await?;
+    let contract_class = provider
+        .get_class(BlockId::Tag(Pending), class_hash)
+        .await?;
+
+    Ok(is_legacy_contract(&contract_class))
+}
+
+async fn extract_class_hash_or_fetch(
+    class_hash: Option<FieldElement>,
+    address: FieldElement,
+    provider: &JsonRpcClient<HttpTransport>,
+) -> Result<FieldElement> {
+    match class_hash {
+        Some(class_hash) => Ok(class_hash),
+        None => get_class_hash_by_address(provider, address)
+            .await?
+            .ok_or_else(|| anyhow!("No class hash exists for the provided address {address:#x}")),
```

Comment:
is it possible? :D Maybe we should rephrase the error message to something like "Failed to fetch class hash for address {}` ?

Reply:
In this case "No class hash exists" actually means that the provided address does not exist in the given network, so maybe let's rephrase it to `The provided address {} does not exist`?

Reply:
After all I decided to remove the `extract_class_hash_or_fetch` function in order to limit RPC calls. All RPC errors related to getting the contract class are now being handled by the `handle_rpc_error` function.